components/constituency_lookup/ConstituencyLookup.tsx

```diff
@@ -49,123 +129,111 @@ const PostcodeLookup = () => {
   // Was hitting into this issue: https://github.com/vercel/next.js/issues/55919
 
   const [formState, setFormState] = useState<FormData>(initialFormState);
-  const [apiResponse, setApiResponse] =
-    useState<ConstituencyLookupResponse | null>(null);
-  const [apiInProgress, setApiInProgress] = useState<boolean>(false);
+  const [apiResponse, setApiResponse] = useState<
+    ConstituencyLookupResponse | false | null
+  >(null);
   const [error, setError] = useState<ErrorCode | null>(null);
 
-  const callApi = async (userPostcode: string, addressSlug?: string) => {
-    // Avoid calling the API multiple time concurrently from fast user input...
-    if (apiInProgress) {
-      console.log("Skipping concurrent API call");
-      return;
+  const [validPostcode, setValidPostcode] = useState<boolean>(false);
+  const [lastValidPcode, setLastValidPcode] = useState<string>("");
+  const [constituencyIdx, setConstituencyIdx] = useState<number | false>(false);
+
+  let userConstituency: Constituency | null = null;
+
+  if (
+    apiResponse &&
+    apiResponse.constituencies &&
+    apiResponse.constituencies.length != 0 &&
+    apiResponse.postcode == lastValidPcode &&
+    constituencyIdx !== false
+  ) {
+    userConstituency = apiResponse.constituencies[constituencyIdx];
+  }
```

Comment:
I _think_ we could use `useMemo` for this with `apiResponse` & `constituencyIdx` as the dependencies, unless that also suffers from the same timing issues you mentioned?

Reply:
So from how I understand the problem useMemo also won't work. 

I think the problem is basically fetch calls can still be waiting to return inside the scope of a ConstituencyLookup component that's now got a stale postcode. So any state they try to check their response against will just be the state that existed when they started? 

Even if there was some way to detect that the component which called the fetch is now stale or even if the postcode is stale it could've switched away and back. 

Probably the 'correct' solution is to hold that state in the parent. But I think this is good enough as I can't really imagine having two of these on the same page.